src/item_factory.cpp

```diff
@@ -636,7 +636,7 @@ void Item_factory::finalize_pre( itype &obj )
 
     if( obj.longest_side == -1_mm ) {
         units::volume effective_volume = obj.count_by_charges() ?
-                                         ( obj.volume / obj.stack_size ) : obj.volume;
+                                         ( obj.volume / ( obj.stack_size != 0 ? obj.stack_size : 1 ) ) : obj.volume;
```

Comment:
Github won't let me make a suggestion here thanks to the line break, but I came across the same issue in #60885 and solved it by checking `obj.count_by_charges() && obj.stack_size > 0`, which I find a little cleaner than your version.